### PR TITLE
Remove netstandard1.5 from some projects

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2278

cc @jskeet

### CODE REVIEW

**Review by jskeet — APPROVED** [review]

**Comment by jtattermusch** [comment]
LGTM once my concern from https://github.com/grpc/grpc-dotnet/issues/2278#issuecomment-1727133647 is addressed.

File: src/Grpc.Core.Api/Grpc.Core.Api.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     <IsGrpcPublishedPackage>true</IsGrpcPublishedPackage>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <TargetFrameworks>net462;netstandard1.5;netstandard2.0;netstandard2.1</TargetFrameworks>
+    <TargetFrameworks>net462;netstandard2.0;netstandard2.1</TargetFrameworks>
```
Comment (jtattermusch) [inline src/Grpc.Core.Api/Grpc.Core.Api.csproj:8]: FYI, `netstandard2.1` isn't listed in the recommended list of frameworks in https://github.com/grpc/grpc-dotnet/issues/2278 (which makes this PR only a partial fix), so I'm not sure if not updating it was intentional or not.

Comment (jskeet) [reply src/Grpc.Core.Api/Grpc.Core.Api.csproj:8]: That may just be my fault when reporting the issue. I'm personally okay with it supporting netstandard2.1 as well, as that doesn't add any burden on dependencies (which need to support netstandard2.0 anyway).

**Comment by JamesNK** [comment]
@jtattermusch Approve please

**Review by jtattermusch — APPROVED** [review]

**Pull request merged**